heartbeat/monitors/wrappers/monitors.go

```diff
@@ -0,0 +1,95 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package wrappers
+
+import (
+	"time"
+
+	"github.com/elastic/beats/heartbeat/eventext"
+	"github.com/elastic/beats/heartbeat/look"
+	"github.com/elastic/beats/heartbeat/monitors/jobs"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// WrapCommon applies the common wrappers that all monitor jobs get.
+func WrapCommon(js []jobs.Job, id string, name string, typ string) []jobs.Job {
+	return jobs.WrapAll(
+		js,
+		monStatus,
+		monTiming,
+		monMeta(id, name, typ),
+	)
+}
+
+func monMeta(id string, name string, typ string) jobs.JobWrapper {
+	return func(job jobs.Job) jobs.Job {
+		return WithFields(
+			common.MapStr{
+				"monitor": common.MapStr{
+					"id":   id,
+					"name": name,
+					"type": typ,
+				},
+			},
+			job,
+		)
+	}
+}
+
+// monStatus wraps the given Job's execution such that any error returned
+// by the original Job will be set as a field. The original error will not be
+// passed through as a return value. Errors may still be present but only if there
+// is an actual error wrapping the error.
+func monStatus(origJob jobs.Job) jobs.Job {
+	return func(event *beat.Event) ([]jobs.Job, error) {
+		cont, err := origJob(event)
+		fields := common.MapStr{
+			"monitor": common.MapStr{
+				"status": look.Status(err),
+			},
+		}
+		if err != nil {
+			fields["error"] = look.Reason(err)
+		}
+		eventext.MergeEventFields(event, fields)
+		return cont, nil
+	}
+}
+
+// monTiming executes the given Job, checking the duration of its run and setting
+// its monStatus.
+// It adds the monitor.duration and monitor.monStatus fields.
```

Comment:
Does it add the status?

Reply:
Ah, not anymore! I refactored that into `monStatus`. I'll fix the comment. They used to be combined.